ofctrl/ofctrl.go

```diff
@@ -290,8 +294,13 @@ func (c *Controller) handleConnection(conn net.Conn) {
 				if c.connectMode == ClientMode {
 					reConnChan = c.connCh
 				}
-				NewSwitch(stream, m.DPID, c.app, reConnChan, c.id)
-
+				s := NewSwitch(stream, m.DPID, c.app, reConnChan, c.id)
+				if err := s.switchConnected(); err != nil {
+					log.Errorf("failed to initialize OpenFlow switch %s: %v", m.DPID, err)
+					s.switchDisconnected(false)
```

Comment:
Reconnection event is sent to Controller when this function returns, so do not need to send this event inside `switchDisconnected`